build.zig

```diff
@@ -150,9 +152,8 @@ pub fn build(b: *std.build.Builder) void {
         lint_tigerstyle_step.dependOn(&run_cmd.step);
 
         // lint_shellcheck
-        const lint_shellcheck = b.addSystemCommand(&.{ "sh", "-c", "command -v shellcheck >/dev/null" ++
-            " || (echo -e '\\033[0;31mPlease install shellcheck - https://www.shellcheck.net/\\033[0m' && exit 1)" ++
-            " && shellcheck $(find ./src ./scripts -type f -name '*.sh')" });
+        var lint_shellcheck = allocator.create(ShellcheckStep) catch unreachable;
+        lint_shellcheck.* = ShellcheckStep.init(allocator);
         const lint_shellcheck_step = b.step("lint_shellcheck", "Run shellcheck on **.sh");
```

Comment:
I don't like how enterprisey doing anything custom in build.zig looks. I _think_ this all can be condensed down to something like

```zig
        // lint_shellcheck
        const lint_shellcheck_step = custom_step(b, "lint_shellcheck", "Run shellcheck on **.sh", allocator, struct {
            fn make(allocator: std.mem.Allocator, step: *std.build.Step) anyerror!void {
                const self = @fieldParentPtr(ShellcheckStep, "step", step);

                var shell = try Shell.init(self.gpa);
                defer shell.deinit();

                if (!shell.exec_status_ok("shellcheck --version")) {
                    shell.echo("{ansi-red}Please install shellcheck - https://www.shellcheck.net/{ansi-reset}", .{});
                    return error.NoShellcheck;
                }

                const scripts = try shell.find(.{
                    .where = &.{ "src", "scripts" },
                    .ends_with = ".sh",
                });

                try shell.exec("shellcheck {scripts}", .{
                    .scripts = scripts,
                });
            }
        });
```

Haven't gone this path just yet --- there's a bunch of breaking changes in build.zig for 0.11, so it might be premature to build thick abstractions here.